paasta_tools/cli/cmds/spark_run.py

```diff
@@ -364,19 +398,25 @@ def get_smart_paasta_instance_name(args):
         return f"{args.instance}_{get_username()}_{how_submitted}"
 
 
-def get_spark_env(args, spark_conf, spark_ui_port, access_key, secret_key):
+def get_spark_env(
+    args: argparse.Namespace,
+    spark_conf: Dict[str, str],
+    aws_creds: Tuple[Optional[str], Optional[str], Optional[str]],
+) -> Dict[str, str]:
+    """Create the env config dict to configure on the docker container"""
+
+    spark_conf_str = create_spark_config_str(spark_conf, is_mrjob=args.mrjob)
     spark_env = {}
 
-    if access_key is not None:
+    access_key, secret_key, _ = aws_creds
+    if access_key:
         spark_env["AWS_ACCESS_KEY_ID"] = access_key
         spark_env["AWS_SECRET_ACCESS_KEY"] = secret_key
         spark_env["AWS_DEFAULT_REGION"] = args.aws_region
     spark_env["PAASTA_LAUNCHED_BY"] = get_possible_launched_by_user_variable_from_env()
-    spark_env["PAASTA_INSTANCE_TYPE"] = "spark"
```

Comment:
I take it the servic_configuration_lib sets this now?

Reply:
oh, nice catch!  I found that this is for the driver environment, which is not configured in the service_configuration_lib.   
This env doesn't seems to impact the actual spark run. However, I've launched the paasta spark-run and did a comparison. The updated PR is now ensure we have exactly the same environment variable on driver.